src/main/java/com/codefellowship/codefellowship/controllers/AppUserController.java

```diff
@@ -79,47 +87,73 @@ public String getTestPage(Model m, Principal p) {
         if(p != null) {
             String username = p.getName();
             AppUser user = appUserRepo.findByUsername(username);
+
             if(user != null) {
                 m.addAttribute("username", user.getUsername());
             }
         }
-        return "test.html"
+
+        return "test.html";
     }
 
-    @GetMapping("/user/{id}")
-    public String getUserInfoPage(Model m, Principal p, @PathVariable long id) {
-        if (p != null) {
-            String username = p.getName();
-            AppUser browsing = appUserRepo.findByUsername(username);
-            m.addAttribute("username", browsingUser.getUsername());
+    @GetMapping("/myprofile")
+    public String getMyProfile(Model m, Principal p) {
+        if(p != null) { //not strictly needed if WebSecurityConfig is set up properly
+            AppUser user = appUserRepo.findByUsername(p.getName());
+            m.addAttribute("user", user);
+            m.addAttribute("username", user.getUsername());
+            return "myprofile";
```

Comment:
This is going to try to return a template named "myprofile.html", but you don't have one. Do you want to return "profile.html" or "userprofile.html" here?